lisa/sut_orchestrator/baremetal/cluster/cluster.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from typing import Any, Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.sut_orchestrator.baremetal.build.build import Build, BuildSchema
+from lisa.sut_orchestrator.baremetal.cluster.readychecker import ReadyCheckerSchema
+from lisa.util import InitializableMixin, field_metadata, subclasses
+from lisa.util.logger import get_logger
+from lisa.util.subclasses import Factory
+
+
+@dataclass_json()
+@dataclass
+class Client:
+    management_port: Optional[int] = field(default=-1)
+    ip_address: Optional[str] = field(default="")
+    username: Optional[str] = field(default="")
+    password: Optional[str] = field(default="")
+    private_key_file: Optional[str] = field(default="")
+    port: Optional[int] = field(default=22)
+    iso_http_url: Optional[str] = field(default="")
+
+
+@dataclass_json()
+@dataclass
+class ClusterSchema(schema.TypedSchema, schema.ExtendableSchemaMixin):
+    type: str = field(default="rackmanager", metadata=field_metadata(required=True))
+    build: Optional[BuildSchema] = None
+    ready_checker: Optional[ReadyCheckerSchema] = None
+    ip_getter: Optional[ReadyCheckerSchema] = None
+    client: Optional[Client] = None
+
+
+class Cluster(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: ClusterSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.cluster_runbook: ClusterSchema = self.runbook
+        self._log = get_logger("cluster", self.__class__.__name__)
+
+    def initialize(self, *args: Any, **kwargs: Any) -> None:
+        source_path = kwargs.pop("source_path", "")
+        if source_path:
+            build_factory = Factory[Build](Build)
+            if self.cluster_runbook.build:
+                build = build_factory.create_by_runbook(self.cluster_runbook.build)
+                build.initialize(
+                    source_path=source_path,
+                    files=self.cluster_runbook.build.files,
+                )
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "cluster"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return ClusterSchema
+
+    def deploy(self) -> Any:
```

Comment:
The `deploy` method should accept an environment parameter. Each node may bring information like management port of rack manager.